### DISCUSSION TITLE
HomeAssistant impostazione porte su Fastgate

### DISCUSSION THREAD

**Discussion by firefox11** [indomus/forum#3023] [HUB: Home Assistant]
Buongiorno a tutti, ho iniziato a prendere confidenza con Home assistant installato su Windows tramite VirtualBox, prima di investire su un raspberry, e mi trovo in difficoltà nel gestire l’accesso a h.a. fuori dalla LAN.
Ho istallato DuckDNS e NGINX, generato il dominio con DuckDNS e settato il tutto su h.a. poi sono andato ad aprire la porta 443 sull’IP di H.A. ma non va. Credo che il problema sia nell’impostazione del forwarding sul modem, con cui non ho molta confidenza. Qualcuno sa darmi una mano? 

Grazie a tutti in ogni caso

**Comment by 2Max2Pax** [comment]
Ciao,
hai chiesto a fastweb che il tuo indirizzo IP sia pubblico?
Non serve che sia statico, ma per essere raggiungibile attraverso duckdns deve essere pubblico. 
Fatweb non fa problemi, a darti l'indirizzo IPv4 pubblico, basta che gli dici che hai le telecamere di videosorveglianza per l'antifurto.

**Reply by firefox11** [reply]
Ciao 2Max, grazie per il riscontro.
Sinceramente avevo dato per scontato che fosse pubblico, ma probabilmente non lo è. Verifico.
Una domanda da non esperto, ma se fosse statico non potrei evitare l’utilizzo di DuckDns? E sai dirmi come mai non funziona con l’ip statico? 
Comunque domani chiamo Fastweb ma rimane sempre da mappare le porte.
A quel punto basta aprire la porta dell’https 443 sull’ip di Home Assistant?

**Reply by 2Max2Pax** [reply]
Funziona anche con l'IP statico, l'importante e' che sia pubblico.
Se fosse statico potresti evitare di usare DuckDns.
La guida su indomus e' chiara e a prescindere che l'indirizzo IP sia statico oppure dinamico e' vivamente consigliato usare un ddns (nel tuo caso DuckDns) ed un reverse proxy (nel tuo caso NGINX) per limitare gli accessi "indesiderati" alla tua rete/computer.
Per quanto riguarda la porta, credo che tu non debba aprire la 443, ma non sono pratico di macchine virtuali. Quando ti colleghi in locale che porta usi?

**Reply by firefox11** [reply]
Ok, chiaro. 
quando mi collego in locale uso la porta 8123, l’indirizzo classico di H.A. http://homeassistant.local:8123/

**Reply by trebestie** [reply]
Non riesco a mettere a fuoco la configurazione
http://homeassistant.local:8123 è una connessione diretta o tramite reverse proxy?

**Reply by firefox11** [reply]
La connessione http://homeassistant.local:8123/ è diretta, è l’indirizzo con cui ci si collega a home assistant una volta istallato. Io sto cercando di collegarmi al programma fuori dalla
LAN usando duckDNS e nginx ( che credo funzioni da reserve proxy) ma il problema credo sia stato centrato da 2max. Fastweb da indirizzi ip privati e quindi con la procedura classica non riesco a raggiungerlo dall’esterno, ma ho trovato quest’articolo su indomus che forse è il caso mio https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-via-reti-provider-tradizionali-o-che-usino-nat-via-cloudflare/ , anche se ancora non ho avuto tempo di leggerlo e di provare ad applicare quanto scritto.
Oppure, come suggerito da 2Max,  chiedo l’indirizzo pubblico a Fastweb e mi complico meno la vita seguendo quanto fatto con duckDns nginx, ma non essendo esperto devo sempre capire se l’impostazione delle porte del modem che faccio è corretta.

**Reply by trebestie** [reply]
Se segui la guida che hai linkato non hai bisogno di aprire porte sul modem.
Se invece vuoi un approccio più classico, e meno sicuro, devi assicurarti di conoscere sempre l'indirizzo pubblico del tuo router e poi definire quale porta aprire e dove redirigere la connessione.
Sul fastgate vai in avanzate -> configurazione manuale porte -> port mapping -> associa nuovo portmapping 
e scrivi qualcosa del genere 
![2024-04-09-100133_897x399_scrot](https://github.com/indomus/forum/assets/17794411/05647792-7de5-4340-95ff-1a930712582f)
a questo punto dall'esterno http://tuoindirizzo.ip.pubblico:9999 ti manda sul tuo home assistant

**Comment by firefox11** [comment]
Grazie 3bestie, appena riesco provo a seguire la guida, sembra anche a me la strada migliore. In ogni caso, a prescindere da quanto farò, mi avete aiutato ad avere le idee più chiare. 
Grazie
Vi aggiorno sui risultati 😀

**Comment by firefox11** [comment]
Ciao a tutti,
ho seguito la guida sotto:
https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-via-reti-provider-che-usino-nat/

putroppo mi sono dovuto fermare all'Aggiornamento dominio (FQND) perchè dopo aver fatto tutta la procedura descritta al link sotto, l'esito del test è KO

https://indomus.it/guide/aggiornare-il-proprio-record-su-duckdns-tramite-linux-debian/

avete idea di quale errore posso aver commesso?
L'unica cosa da modificare è la stringa in cui va messo il nome del dominio, il token e l'indirizzo IP del modem.
L'unico dubbio può essere sull'IP, le altre cose l'ho copiate :(

**Reply by firefox11** [reply]
Specifico che come ip non ho messo quello del modem ma quello esterno del server virtuale, forse non è quello da inserire nella stringa?

**Reply by snakuzzo** [reply]
Ti sconsiglio di usare quel metodo con la VM ponte su GCP.
Investi qualche euro e usa il tunnel gratuito con cloudflare

https://indomus.it/guide/configurare-cloudflare-per-il-proprio-dominio-personale-per-protezione-reverse-proxy-cdn-e-altro/

https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-via-reti-provider-tradizionali-o-che-usino-nat-via-cloudflare/

**Reply by firefox11** [reply]
Ciao Snakuzzo,

grazie per il suggerimento, proverò a seguire il link inviato.
Per mia cultura, non essendo un grande esperto, come mai sconsigli il primo metodo seguito?
Tra l’altro al momento non riesco a farlo funzionare, anche se mi piacerebbe capire come mai non va :)

**Reply by snakuzzo** [reply]
1. Da quello che leggo sei poco pratico...e quello non è di certo il modo più semplice per fare quello che ti serve
2. Ti offre meno possibilità di quelle offerte dal tunnel di Cloudflare
3. La VM su Google ha delle limitazioni (in termini proprio di quantità) a livello di traffico...se sfori paghi. Così come potresti pagare se cambia qualcosa lato GCP

**Reply by firefox11** [reply]
Sicuramente non sono molto pratico, ma mi piace approfondire. 
Oltre alla curiosità e al piacere, credo sia importante farlo quando ci si addentra in cose poco conosciute.
Se hai qualche suggerimento (link utili da leggere, video da guardare, etc..) sono ben accetti. 

Nel frattempo proverò a portare a termine la procedura che hai suggerito.


ti ringrazio
Buona giornata

**Reply by snakuzzo** [reply]
La nostra guida per passare da una VM su GCP spiega già tutto nei minimi dettagli.
Stesso dicasi per la guida relativa a Cloudflare

**Reply by firefox11** [reply]
Seguito la procedura e funziona tutto, anche se il grado di difficoltà dell'altra procedura non mi è sembrato così dissimile da questa seconda modalità.
In ogni caso questa funziona.
Ho  provato a collegare il link anche tramite l'app di HA ma non sembra funzionare.
La procedura fatta è compatibile con l'utilizzo della sua app tramite azioni aggiuntive, oppure no?.

vi ringrazio ancora

**Reply by snakuzzo** [reply]
Beh! Se non ti risulta così dissimile, c'è qualche problema di fondo :)
Scherzi a parte, le due guide sono totalmente differenti, sia a livello di complessità che proprio di "sostanza", anche se effettivamente sfruttano lo stesso concetto del tunnel.
Non ho capito cosa intendi per "collegare il link anche tramite app di HA". Non mi puoi venire a parlare di compatibilità solo per aver introdotto il tunnel di Cloudflare. Se fosse stato "incompatibile" (termine usato totalmente fuori luogo) non avremmo nemmeno fatto le guide. Se non funziona, ribatti tutto e controlla

**Reply by firefox11** [reply]
Non so dove possa essere il problema però con le guide che fate, essendo fatte bene, la difficoltà nel seguire le due procedure non mi sembra diversa. 
Altra cosa è avere le competenze per poter capire fino in fondo cosa si stia facendo e per poter fare delle modifiche in autonomia.

In ogni caso ho riavviato tutto e ora funziona anche l’app 🎉

perfetto 

grazie

**Reply by snakuzzo** [reply]
Comunque le guide non sono fatte per essere seguite tipo copia-incolla e basta. Il punto è proprio questo...le guide vanno seguite e capite. Se fai le cose come un manuale di istruzioni, non saprai mai nulla di quello che hai fatto. Per te resterà solo un "sono stato bravo a copiare e incollare i comandi".
Infine, contento che tu abbia risolto, ma il riavvio è proprio il classico esempio di una cosa che si fa quando non si sa dove mettere le mani. Allora uno dice "vabbè mo provo a riavviare" e per botta di culo funziona :)
Considera che, nonostante tutte le prove, le installazioni, e le decine e decine di aggiornamenti che si fanno c'è chi ha tutto up& running da anni

**Reply by firefox11** [reply]
Non sono un programmatore o un informatico, e non me ne faccio una colpa.
Sicuramente il primo passo per capire è praticare le cose e seguire, quindi copiare, chi le fa e le sa meglio di te.
Così facendo, praticando e sbagliando, leggendo e applicando, si riesce a capire di più e ad imparare.
Certo è che se uno non ci mette la testa concordo che serva a poco, ma probabilmente uno che la pensa così non si sbatte neanche ad avvicinarsi a certe cose.
Mi sono avvicinato alla community proprio per cercare di capire ed imparare da chi ne sa di più, altrimenti avrei provato con le varie guide, o videoguide, seguendole pedissequamente fino a trovarne una che riuscivo a far funzionare, oppure compravo una delle tante soluzioni chiavi in mano.

Ho iniziato il post dicendo che sto cercando di prendere confidenza con HA e con altri programmi, per es. Linux che non uso nel mio quotidiano, prima di investire dei soldi nell’acquisto delle componenti, proprio perché voglio provare a capire, ed imparare ad utilizzare e gestire, tutto il più possibile.
Mi muovo per steps proprio per comprendere meglio.
Credo che sia alla mia portata anche se ci vorrà del tempo, se poi capisco che non lo è me ne farò una ragione.

A tal proposito sto ancora cercando di capire cosa sbaglio nell’applicare la prima soluzione, non tanto perché mi serva, ma perché mi piacerebbe capire e farla funzionare, poco importa se non è la soluzione ottimale.
Al fine d’imparare e capire credo che sia importante provare strade alternative, così, oltre ad imparare e vedere soluzioni diverse, si capiscono meglio le differenze ed il perché una strada è meglio dell’altra.

Ma alla richiesta di suggerimenti nel risolvere la problematica sul metodo con la VM ponte su GCP, hai dato una risposta che non mi sembra tanto in linea con l’ultimo post che hai scritto, anzi, credo che porti più nella direzione del copia e incolla.
Alla mia richiesta di suggerimenti per cercare di capire meglio come applicarla hai risposto che :
“la nostra guida per passare da una VM su GCP spiega già tutto nei minimi dettagli.”

Posso capire che non si abbia voglia di spiegare nei minimi dettagli, o di star dietro a chi ne sa meno anche se a voglia di approfondire ed imparare, ma se è così quantomeno evitiamo le prediche e l’ipocrisia.
Se invece l’intento è quello di spronare a fare meglio, ribadisco che sono qui per imparare, ma se si vuole insegnare anche chi impartisce la lezione si deve applicare.
Detto questo continuo con il mio processo d’apprendimento e se qualcuno vorrà, o vorrai, dare dei suggerimenti sono i ben accetti.

grazie ancora

**Reply by snakuzzo** [reply]
Il fatto è che nella guida che sfrutta la VM su GCP è spiegato tutto in dettaglio. Ok la community, ok l'aiuto, ma da come eri partito all'inizio e con quello che avevi scritto, risulta troppo complicato darti supporto per una configurazione che non è proprio tra le più semplici.
Il mio consiglio è anche il nostro consiglio (di indomus) non solo per una questione di semplicità operativa (e tu forse sei l'unico ad aver detto che non c'è tutta questa differenza), ma anche per una questione di sicurezza e di possibilità messe a disposizione. Detto questo, non è che si può spiegare tutto nei minimi dettagli ad ogni richiesta di supporto e questo è il motivo per cui ci sono i nostri articoli e guide. Cerchiamo di farli più dettagliati possibile proprio per consentire a chi li legge (esperti e non) di poter capire ed eventualmente approfondire in autonomia.
Questo per dirti che la mia non era "rottura di balle nel risponderti e spiegarti le cose", ma noi non è che siamo degli insegnanti, né possiamo e/o pretendiamo di esserlo. Questa è una community ed è di tutti. Chi vuole può anche dedicarti 24h al giorno per farti imparare le cose, ma devi sapere che non è che siamo a scuola dove per ogni minima cosa si chiede al maestro/professore. Qui, se non si è pratici, tocca mettersi lì, cercare, leggere tanto e sbattere ore e ore la testa. Le cose si apprendono in questo modo...nessuno "nasce imparato" e noi sproniamo la gente a fare così, ribadendo che le guide, per quanto precise possano essere, devono essere un input per iniziare ad apprendere e muoversi per conto proprio e non per avere la soluzione chiavi in mano a scatola chiusa. Quando leggi, se non sai di che cacchio si sta parlando, al posto di chiedere in chat, forum o altro, si va su un motore di ricerca e si cerca quello che non si è capito (errori, comandi, etc)...la richiesta esplicita di aiuto deve essere solo l'ultimo step...altrimenti diventa tutto inutile e vuol dire che noi (indomus) stiamo solo perdendo tempo.

P.S. I suggerimenti ti sono stati dati ed applicarsi ora sulla soluzione ponte su GCP non ha alcun senso logico (per quanto detto in precedenza). Piuttosto, cerca di apprendere bene il funzionamento del tunnel di Cloudflare.

**Reply by firefox11** [reply]
Rispondi senza leggere quello che scrivo.
Nessuno vuole la spiegazione nel minimo dettaglio, e quanto scrivi è quello che faccio tutti i giorni.
Tant'è che ti ho chiesto dove andare a leggere le cose e non di spiegarmele.
rileggi qualche post sopra e troverai scritto:

"Sicuramente non sono molto pratico, ma mi piace approfondire.
Oltre alla curiosità e al piacere, credo sia importante farlo quando ci si addentra in cose poco conosciute.
Se hai qualche suggerimento (link utili da leggere, video da guardare, etc..) sono ben accetti."

Non trovo coerenza con quanto scrivi, trovo solo narcisismo

**Reply by snakuzzo** [reply]
Ah sì ?
Vabbè, me ne farò una ragione.
Ciao.

P.S. A saperlo mi facevo i ca**i miei :)

**Reply by xmaxtclub** [reply]
> Rispondi senza leggere quello che scrivo. Nessuno vuole la spiegazione nel minimo dettaglio, e quanto scrivi è quello che faccio tutti i giorni. Tant'è che ti ho chiesto dove andare a leggere le cose e non di spiegarmele. rileggi qualche post sopra e troverai scritto:
> 
> "Sicuramente non sono molto pratico, ma mi piace approfondire. Oltre alla curiosità e al piacere, credo sia importante farlo quando ci si addentra in cose poco conosciute. Se hai qualche suggerimento (link utili da leggere, video da guardare, etc..) sono ben accetti."
> 
> Non trovo coerenza con quanto scrivi, trovo solo narcisismo

buongiorno, ha trovato poi una soluzione senza dover essere un genio programmatore e co. ? mi hanno cambiato operatore ieri, da wind a fastweb, e ovviamente non mi funziona più l'accesso da remoto ad h.a. anche dopo aver riconfigurato il router con gli stessi parametri del router precedente. grazie

**Reply by indomus** [reply]
apri (se credi) un thread apposito, fornendo maggiori dettagli. grazie
